RecoTracker/FinalTrackSelectors/plugins/DuplicateTrackMerger.cc

```diff
@@ -104,41 +97,56 @@ using namespace reco;
+     desc.add<double>("maxDCA",30.0);
+     desc.add<double>("maxDPhi",0.30);
+     desc.add<double>("maxDLambda",0.30);
+     desc.add<double>("maxDdsz",10.0);
+     desc.add<double>("maxDdxy",10.0);
+     desc.add<double>("maxDQoP",0.25);
+     desc.add<std::string>("forestLabel","MVADuplicate");
+     desc.add<bool>("useInnermostState",true);
+     desc.add<std::string>("ttrhBuilderName","WithAngleAndTemplate");
+     descriptions.add("DuplicateTrackMerger", desc);
+}
+
+  
+DuplicateTrackMerger::DuplicateTrackMerger(const edm::ParameterSet& iPara) : forest_(nullptr), merger_(iPara)
+{
+
   useForestFromDB_ = true;
 
-  minDeltaR3d_ = -4.0;
-  minBDTG_ = -0.96;
-  minpT_ = 0.2;
-  minP_ = 0.4;
-  maxDCA_ = 30.0;
-  maxDPhi_ = 0.30;
-  maxDLambda_ = 0.30;
-  maxDdsz_ = 10.0;
-  maxDdxy_ = 10.0;
-  maxDQoP_ = 0.25;
-  if(iPara.exists("minpT"))minpT_ = iPara.getParameter<double>("minpT");
-  if(iPara.exists("minP"))minP_ = iPara.getParameter<double>("minP");
-  if(iPara.exists("maxDCA"))maxDCA_ = iPara.getParameter<double>("maxDCA");
-  if(iPara.exists("maxDPhi"))maxDPhi_ = iPara.getParameter<double>("maxDPhi");
-  if(iPara.exists("maxDLambda"))maxDLambda_ = iPara.getParameter<double>("maxDLambda");
-  if(iPara.exists("maxDdsz"))maxDdsz_ = iPara.getParameter<double>("maxDdsz");
-  if(iPara.exists("maxDdxy"))maxDdxy_ = iPara.getParameter<double>("maxDdxy");
-  if(iPara.exists("maxDQoP"))maxDQoP_ = iPara.getParameter<double>("maxDQoP");
-  if(iPara.exists("source"))trackSource_ = consumes<reco::TrackCollection>(iPara.getParameter<edm::InputTag>("source"));
-  if(iPara.exists("minDeltaR3d"))minDeltaR3d_ = iPara.getParameter<double>("minDeltaR3d");
-  if(iPara.exists("minBDTG"))minBDTG_ = iPara.getParameter<double>("minBDTG");
+  trackSource_ = consumes<reco::TrackCollection>(iPara.getParameter<edm::InputTag>("source"));
+  minDeltaR3d2_ = iPara.getParameter<double>("minDeltaR3d"); minDeltaR3d2_*=std::abs(minDeltaR3d2_);
+  minBDTG_ = iPara.getParameter<double>("minBDTG");
+  minpT2_ = iPara.getParameter<double>("minpT"); minpT2_ *= minpT2_;
+  minP_ = iPara.getParameter<double>("minP");
+  maxDCA2_ = iPara.getParameter<double>("maxDCA"); maxDCA2_*=maxDCA2_;
+  maxDPhi_ = iPara.getParameter<double>("maxDPhi");
+  maxDLambda_ = iPara.getParameter<double>("maxDLambda");
+  maxDdsz_ = iPara.getParameter<double>("maxDdsz");
+  maxDdxy_ = iPara.getParameter<double>("maxDdxy");
+  maxDQoP_ = iPara.getParameter<double>("maxDQoP");
 
   produces<std::vector<TrackCandidate> >("candidates");
   produces<CandidateToDuplicate>("candidateMap");
 
-  gbrVals_ = new float[9];
   dbFileName_ = "";
-  if(iPara.exists("GBRForestLabel"))forestLabel_ = iPara.getParameter<std::string>("GBRForestLabel");
+  forestLabel_ = iPara.getParameter<std::string>("forestLabel");
+
   if(iPara.exists("GBRForestFileName")){
-    dbFileName_ = iPara.getParameter<std::string>("GBRForestFileName");
+    dbFileName_ = iPara.getUntrackedParameter<std::string>("GBRForestFileName");
```

Comment:
it is used only in testing, production uses db
we can track, default to "" an set useForestFromDB_ = dbFileName.empy()
?
possible